fluss-common/src/main/java/com/alibaba/fluss/types/MultisetType.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.fluss.types;
+
+import com.alibaba.fluss.annotation.PublicEvolving;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+import static com.alibaba.fluss.utils.Preconditions.checkNotNull;
+
+/**
+ * Data type of a multiset (=bag). Unlike a set, it allows for multiple instances for each of its
+ * elements with a common subtype. Each unique value (including {@code NULL}) is mapped to some
+ * multiplicity. There is no restriction of element types; it is the responsibility of the user to
+ * ensure uniqueness.
+ *
+ * <p>A conversion is possible through a map that assigns each value to an integer multiplicity
+ * ({@code Map<t, Integer>}).
+ *
+ * @since 0.8
+ */
+@PublicEvolving
+public class MultisetType extends DataType {
```

Comment:
Please remove the `MultisetType` and `VarBinaryType` and `VarCharType`. 
 - MultisetType can be represented as MapType, and we should support MapType first, otherwise the `MultisetType` is not able to be used.
 - We already have `BytesType` and `StringType` which can represent the `VarBinaryType` and `VarCharType`. 
 - Variable-length types often require the storage layer to enforce length constraints (e.g., `VARCHAR(255)`), which is not currently supported by Fluss. Given that such constraints are rarely used in analytical workloads, the practical value of these types is limited in our target scenarios.